solr/core/src/java/org/apache/solr/rest/ManagedResourceStorage.java

```diff
@@ -91,7 +91,7 @@ public static StorageIO newStorageIO(String collection, SolrResourceLoader resou
       zkClient = ((ZkSolrResourceLoader)resourceLoader).getZkController().getZkClient();
       try {
         zkConfigName = ((ZkSolrResourceLoader)resourceLoader).getZkController().
-            getZkStateReader().readConfigName(collection);
+            getZkStateReader().getClusterState().getCollection(collection).getConfigName();
```

Comment:
BTW for brevity, you can remove `getZkStateReader().` here and elsewhere since ZkController has a convenience method for the cluster state.